server/keyshare/myirmaserver/server.go

```diff
@@ -311,10 +311,10 @@ func (s *Server) handleEmailLogin(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// In case sendLoginEmail fails with errEmailNotFound or errTooManyRequests, then we
+	// In case sendLoginEmail fails with errEmailNotFound, errTooManyRequests or ErrInvalidEmailDomain, then we
 	// should not write an error. Otherwise, we would leak information about our user base.
 	err := s.sendLoginEmail(r.Context(), request)
-	if err != nil && err != errEmailNotFound && err != errTooManyTokens {
+	if err != nil && err != errEmailNotFound && err != errTooManyTokens && err != keyshare.ErrInvalidEmailDomain {
```

Comment:
What's exactly the issue about returning a 400 when the domain is not valid?

Reply:
Double checked and you're right indeed! It does not cause an issue (in the way of leaking information) if we would return the 400 at a properly formatted but non-working email address. I'll reverse that bit and correct the test accordingly.